repairnator/src/main/java/fr/inria/spirals/repairnator/process/step/GatherTestInformation.java

```diff
@@ -151,12 +153,24 @@ public FileVisitResult preVisitDirectory(Path file, BasicFileAttributes attrs) t
         }
 
         if (this.getState() == ProjectState.HASTESTFAILURE) {
-            this.shouldStop = false;
+        	if (!(inspector.getMode() == RepairMode.FORBEARS)) {
+        		this.shouldStop = false;
+        	} else {
+        		this.shouldStop = true;
+        	}
         } else if (this.getState() == ProjectState.HASTESTERRORS) {
             this.addStepError("Only get test errors, no failing tests. It will try to repair it.");
-            this.shouldStop = false;
+            if (!(inspector.getMode() == RepairMode.FORBEARS)) {
+        		this.shouldStop = false;
+        	} else {
+        		this.shouldStop = true;
+        	}
         } else {
-            this.shouldStop = true;
+        	if (!(inspector.getMode() == RepairMode.FORBEARS)) {
+        		this.shouldStop = true;
+        	} else {
+        		this.shouldStop = false;
+        	}
```

Comment:
You should simplify this very easily: you always take the contrary of my shouldStop status. 
Then add a line at the end like:
```java
this.shouldStop = (inspector.getMode() == RepairMode.FORBEARS) ? !this.shouldStop : this.shouldStop;
```